api/v1alpha1/oauth2client_types.go

```diff
@@ -145,12 +145,17 @@ type OAuth2ClientSpec struct {
 	// Audience is a whitelist defining the audiences this client is allowed to request tokens for
 	Audience []string `json:"audience,omitempty"`
 
-	// +kubebuilder:validation:Pattern=([a-zA-Z0-9\.\*]+\s?)+
+	// +kubebuilder:validation:Pattern=([a-zA-Z0-9\.\*]+\s?)*
 	//
-	// Scope is a string containing a space-separated list of scope values (as
+	// [DEPRECATED] Scope is a string containing a space-separated list of scope values (as
```

Comment:
we can use the kubebuilder annotation here
```
// +kubebuilder:deprecatedversion
warningstring
marks this version as deprecated.
```

Reply:
Are you sure this works? I tried that but it had no effect on the generated manifest. I have a feeling this only works on the entire config object, not a single field.

I optimistically changed it anyway.

Reply:
Ok i might have misread that 🤔 
```
//+kubebuilder:deprecatedversion:warning="example.com/v1alpha1 CronTab is deprecated" 
```
this might be set on the whole apiversion, and not a single field in the spec, as the openapi schema allows for field deprecation, but kubebuilder may not yet have the tooling for this 😞